### PR TITLE
Update product-categories.js to show the Edit Category modal

### PR DESCRIPTION
**Pull request by deivamagalhaes**
# Summary

This PRs adds the JS used to show a modal if the Google product category ID is changed for an existing product category.

## Details

The JS here was "inspired" by the one used for the Settings modal. I've opted for creating new methods for the message and buttons because it seems the message for the Settings screen was updated on the pitch, so I'm guessing we may want to change that one separately in the near future.

I also took the opportunity to fix loading the saved Google product category ID for an existing category.

## UI Changes

A modal is shown if the Google product category ID is changed for an existing product category: https://cloud.skyver.ge/6que6PK6

### Setup

1. Add `add_filter( 'wc_facebook_commerce_is_available', '__return_true' )`
1. Add `add_filter( 'wc_facebook_commerce_is_connected', '__return_true' )`

### Steps

1. Go to Products > Categories
1. Add a new category, selecting a Google product category for it
1. Click "Add new category"
    - [x] The modal is not displayed
    - [x] The form is submitted
1. Edit the existing category
1. Select a Google product category for it
1. Click "Update"
    - [x] The modal is displayed
1. Click "Cancel"
    - [x] The form is _not_ submitted
1. Click "Update" again
    - [x] The modal is displayed
1. Click "Update default Google product category"
    - [x] The form is submitted

## Before merge

- [ ] I have confirmed these changes in each supported minor WooCommerce version

### CODE REVIEW

**Review by wvega — CHANGES_REQUESTED** [review]
@deivamagalhaes looks good and the modal shows up as expected. I left a suggestion to remove a parameter I think we don't need and also have a question:

Should we set a specific width for the select fields?

This is how it looks on:

- Edit Category page with some categories already selected: https://cloud.skyver.ge/P8ubLyn2
- Edit Category page with no categories selected: https://cloud.skyver.ge/xQuLyYPl
- Add Category form: https://cloud.skyver.ge/yAubdZG4

I was going to suggest updating the field JS to set the width, but I think each screen where we show the field is going to use something different. It is `400px` on the Commerce settings (already set by WooCommerce's CSS), the text area in the Add/Edit Category forms use `95%` and it could be different on the Product tabs.

File: includes/Admin/Product_Categories.php [hunk 1]
```diff
@@ -115,8 +115,10 @@ class="button button-large button-primary"
 	 * @internal
 	 *
 	 * @since 2.1.0-dev.1
+	 *
+	 * @param string $taxonomy current taxonomy slug
```
[the remark this answers is not in the document]
Comment (deivamagalhaes) [reply includes/Admin/Product_Categories.php]: Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1570/commits/f84f8a0563be6147d94561dc3e637bbf6d29423e.

File: includes/Admin/Product_Categories.php [hunk 2]
```diff
@@ -115,8 +115,10 @@ class="button button-large button-primary"
...
 	 */
-	public function render_add_google_product_category_field() {
+	public function render_add_google_product_category_field( $taxonomy ) {
```
[the remark this answers is not in the document]
Comment (deivamagalhaes) [reply includes/Admin/Product_Categories.php]: Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1570/commits/f84f8a0563be6147d94561dc3e637bbf6d29423e.

**Review by wvega — APPROVED** [review]
Thank you for the updates @deivamagalhaes. LGTM!

**Pull request merged**